pkg/user/tx_client.go

```diff
@@ -251,18 +289,60 @@ func SetupTxClient(
 		return nil, fmt.Errorf("failed to create signer: %w", err)
 	}
 
-	return NewTxClient(encCfg.Codec, signer, conn, encCfg.InterfaceRegistry, options...)
+	txClient, err := NewTxClient(encCfg.Codec, signer, conn, encCfg.InterfaceRegistry, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Always start the tx queue
+	if txClient.txQueue != nil {
+		if err := txClient.txQueue.start(ctx); err != nil {
+			return nil, fmt.Errorf("failed to start tx queue: %w", err)
+		}
+	}
+
+	return txClient, nil
 }
 
 // SubmitPayForBlob forms a transaction from the provided blobs, signs it, and submits it to the chain.
 // TxOptions may be provided to set the fee and gas limit.
+// This method uses the tx queue infrastructure and blocks until the transaction is confirmed.
 func (client *TxClient) SubmitPayForBlob(ctx context.Context, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error) {
-	resp, err := client.BroadcastPayForBlob(ctx, blobs, opts...)
-	if err != nil {
-		return nil, err
+	if client.txQueue == nil {
+		return nil, errors.New("tx queue not configured")
 	}
 
-	return client.ConfirmTx(ctx, resp.TxHash)
+	if !client.txQueue.started.Load() {
+		return nil, errors.New("tx queue not started")
+	}
+
+	resultsC := make(chan SubmissionResult, 1)
+
+	job := &SubmissionJob{
+		Blobs:    blobs,
+		Options:  opts,
+		Ctx:      ctx,
+		ResultsC: resultsC,
+	}
+
+	client.SubmitJob(job)
+
+	// Block waiting for the result
+	result := <-resultsC
+	if result.Error != nil {
+		return result.TxResponse, result.Error
+	}
+
+	return result.TxResponse, nil
+}
+
+// SubmitJob submits a job to the tx queue for parallel processing
+func (client *TxClient) SubmitJob(job *SubmissionJob) {
+	if client.txQueue != nil {
+		client.txQueue.submitJob(job)
+	} else {
+		job.ResultsC <- SubmissionResult{Error: errors.New("tx queue not configured")}
+	}
```

Comment:
Left this as an easter egg for power users who want to insert their own txs directly into the queue w/o blocking on submit

Reply:
so that we can have a txsim equivalent without having to wrap a SubmitPayForBlob and then recombine all the channels

imo we should still keep this